pkg/host/internal/bridge/ovs/ovs.go

```diff
@@ -0,0 +1,677 @@
+	}
+
+	if iface.Error != nil {
+		funcLog.V(2).Info("getCurrentBridgeState(): interface has an error, remove it from the bridge state", "interface", iface.Name, "error", iface.Error)
+		// interface has an error, do not report info about it to let the operator try to recreate it
+		return currentConfig, nil
+	}
+
+	port, err := o.getPortByInterface(ctx, dbClient, iface)
+	if err != nil {
+		return nil, err
+	}
+	if port == nil {
+		return currentConfig, nil
+	}
+
+	if !bridge.HasPort(port.UUID) {
+		// interface belongs to a wrong bridge, do not include uplink config to
+		// the current bridge state to let the operator try to fix this
+		return currentConfig, nil
+	}
+	currentConfig.Uplinks = []sriovnetworkv1.OVSUplinkConfigExt{{
+		PciAddress: knownConfigUplink.PciAddress,
+		Name:       knownConfigUplink.Name,
+		Interface: sriovnetworkv1.OVSInterfaceConfig{
+			Type:        iface.Type,
+			ExternalIDs: updateMap(knownConfigUplink.Interface.ExternalIDs, iface.ExternalIDs),
+			Options:     updateMap(knownConfigUplink.Interface.Options, iface.Options),
+			OtherConfig: updateMap(knownConfigUplink.Interface.OtherConfig, iface.OtherConfig),
+		},
+	}}
+	return currentConfig, nil
+}
+
+func (o *ovs) getRootObj(ctx context.Context, dbClient client.Client) (*OpenvSwitchEntry, error) {
+	ovsList := []*OpenvSwitchEntry{}
+	if err := dbClient.List(ctx, &ovsList); err != nil {
+		return nil, fmt.Errorf("can't retrieve root object uuid from Open_vSwitch table")
+	}
+	if len(ovsList) == 0 {
+		return nil, fmt.Errorf("Open_vSwitch table is empty")
+	}
+	return ovsList[0], nil
+}
+
+// if the provided context has no timeout, the default timeout will be set
+func setDefaultTimeout(ctx context.Context) (context.Context, context.CancelFunc) {
+	_, ok := ctx.Deadline()
+	if ok {
+		// context already contains deadline,
+		// return original context and dummy cancel function
+		return ctx, func() {}
+	}
+	return context.WithTimeout(ctx, defaultTimeout)
+}
+
+// resulting map contains keys from the old map with values from the new map.
+// if key from the old map not found in the new map it will not be added to resulting map
+func updateMap(old, new map[string]string) map[string]string {
+	result := map[string]string{}
+	for k := range old {
+		val, found := new[k]
+		if found {
+			result[k] = val
+		}
+	}
+	return result
+}
+
+// initialize and return OVSDB client
+func getClient(ctx context.Context) (client.Client, error) {
+	openvSwitchEntry := &OpenvSwitchEntry{}
+	bridgeEntry := &BridgeEntry{}
+	interfaceEntry := &InterfaceEntry{}
+	portEntry := &PortEntry{}
+	clientDBModel, err := DatabaseModel()
+	if err != nil {
+		return nil, fmt.Errorf("can't create client DB model: %v", err)
+	}
+
+	dbClient, err := client.NewOVSDBClient(clientDBModel,
+		client.WithEndpoint(vars.OVSDBSocketPath),
+		client.WithLogger(&log.Log))
+	if err != nil {
+		return nil, fmt.Errorf("can't create DB client: %v", err)
+	}
+
+	err = dbClient.Connect(ctx)
```

Comment:
Maybe. I tried to use single client instance during the initial implementation but discovered that it is hard to handle failures/retries in a controlled manner. I decide to go with the current approach to make sure that we will return an error fast enough in case if OVS become unavailable on the node and can retry from the scratch during the next iteration.